### ISSUE TITLE
Implementations Require Concrete Examples

### ISSUE THREAD

**Issue by neurolabusc** [bids-standard/bids-specification#1350]
Many extensions with to the BIDS format create a wish list of useful sequence details without clear instructions regarding how to implement these. It would really help to have sample DICOM files from the major manufacturers as well as an example of the desired BIDS conversion. This would help automate conversion support, identify manufacturer specific terminology and also help identify missing details that can be addressed by the relevant DICOM working groups.

While this is a general concern, my issue was elicited by [dcm2niix issue 646](https://github.com/rordenlab/dcm2niix/issues/646). @agahkarakuzu you have supplied a [desired BIDS conversion](https://osf.io/k4bs5/), but it does not contain a source DICOM. The current [specification seems underspecified](https://bids-specification.readthedocs.io/en/stable/appendices/qmri.html#tb1epi-specific-notes). It is unclear how to determine stimulated echo (STE) for various manufacturers (GE, Philips) as well as different systems (Siemens VE11 Classic DICOM vs Siemens XA30 enhanced DICOM). I would suggest a model like the [dcm_qa](https://github.com/neurolabusc/dcm_qa) repositories, that provide reference datasets that can aid developers for all the popular conversion tools. 

Ideally, it would be great if new specification generation could happen as a virtuous cycle with conversion software development. This would allow the specification to be refined to capture edge cases. I would be happy to have BIDS extension designers submit pull requests to dcm2niix (C). Alternatively, specification creators could develop dicm2nii (Matlab) or dicom2nifti (Python) if that is their preferred language. A reference dataset and software implementation can help avoid a lot of unintended consequences.

The reference datasets also have other benefits beyond tool conversion. For example, as long as the user has the correct licenses, most Siemens users can `Phoenix` a DICOM sequence from another scanner onto their system to easily clone a sequence.

**Comment by neurolabusc** [comment]
@Remi-Gau I do not agree with the more restricted title. For my specific case of developing DICOM support for BIDS my concerns apply to all medical images (MRI, PET, CT, SPECT, etc). However, I think this applies to other domains like PET and ERP where the manufacturers have their own data formats that must be converted to standards. 

Creating data standards that include a description and an example of the desired structured data without the source original data places a heavy burden on users and knock on effects on developers, and does not benefit from insight gained by explicitly mapping the desired terminology to that used by the manufacturer (e.g. different manufacturers use the DICOM tag `Repetition Time` differently for some sequences).

**Comment by Remi-Gau** [comment]
Ah yes sorry. I was too hasty in the renaming. 

I tried to give your title a slightly less broad scope but I swang a bit too much the other.

In an ideal world I would definitely prefer most impletementations that are BIDS related to come with examples but for the sake of this issue I would at least restrict it to be about conversion implementation.

**Comment by Remi-Gau** [comment]
FWIW I do agree with your feeling on this and I think we should try to push BIDS extension (when they are about raw data) to come with conversion examples.

**Comment by neurolabusc** [comment]
@Remi-Gau I do appreciate your feedback as it helped me fully describe my concerns.

**Comment by agahkarakuzu** [comment]
@neurolabusc I am aware of this problem and I definitely agree on the necessity of providing source files and conversion mappings from the get go. 

I also agree that for certain qMRI acquisitions, BEP001 is a wish list of specifications in its current state. But it was the best first thing we could do to start somewhere. I am trying to support users one request at a time to solve the issue for their specific setup. When developing sequences, I conform to the tags in the wish list and we suggest WIP sequence developers to follow that as well. For a while, it is somewhat a cold start problem, but I hope we'll get there. 

Each qMRI-focused WIP sequence has shortcomings in adding important tags to metadata, e.g., RepetitionTimeExcitation  for MP2RAGE can be found only in the protocol pdf, not in the file header. Then XA30 came and things got even more complicated for most of the sequences. Not to mention that the complexity is multiplied with each vendor. 

For TB1EPI specific issue, I tagged people from our team (in the respective issue) who may have the source data.

**Comment by arokem** [comment]
I am trying to wrap my head around the scope of this issue: what data types and modalities are affected by this? It sounds like at least qMRI measurements are affected by this, but is that the extent of the issue? Or does this affect other MRI measurements (e.g., fMRI/dMRI?) and other modalities? Sounds like the scope of this issue could get quite complicated with the wealth of different human electrophysiology vendors and file formats. Does it hit us there as well?

**Comment by neurolabusc** [comment]
@arokem I think this has been a pervasive issue with modality specific BEPs. When the original BIDS was drafted, @chrisgorgo spent a lot of time coordinating with people like me to work out both what was required for thorough analysis but also what was available from the existing data. However, it seems like subsequent BEPs have been drafted that are based on wish lists of what is required for thorough analysis, but no concrete images are provided and no clear consideration for how these fields can be derived. 

For many of the BEPs, I have tried to make comments during the development to try to correlate proposed BIDS fields with DICOM tags. However, I think implementation considerations need to intregrated with the development of these specifications. 


 - ASL [BEP005](https://docs.google.com/document/d/15tnn5F10KpgHypaQJNNGiNKsni9035GtDqJzWqkkP6c/edit) I installed all the product and research sequences I am aware of that are compatible with our Siemens instrument for the [dcm_qa_asl](https://github.com/neurolabusc/dcm_qa_asl) validation dataset. For Siemens classic V* series @machappell helped describe [parameters from the CSA header](https://github.com/rordenlab/dcm2niix/tree/master/BIDS#manufacturer-siemens-arterial-spin-labeling), though support for enhanced XA remains largely undocumented (with the exception of [2D product sequences](https://github.com/neurolabusc/dcm_qa_xa30)).  I worked with @mr-jaemin to document [GE product sequences](https://github.com/rordenlab/dcm2niix/tree/master/GE#arterial-spin-labeling) but I know of no exemplars of research sequences. I did work with @HenkMutsaerts and @jan-petr for Philips support (with help from Philips clinical scientists @drmclem and @sandeepganji), though these seem underspecified.
 - PET [BEP009](https://docs.google.com/document/d/1mqMLnxVdLwZjDd4ZiWFqjEAmOmfcModA_R535v3eQs0/edit#heading=h.4k1noo90gelw) I made suggestions during the draft and worked with @CPernet so [dcm2niix can better support PET2BIDS](https://github.com/openneuropet/PET2BIDS), met with DICOM WG-03 members. While ECAT validation datasets are provided, there are no DICOM validation datasets. One crucial issue with PET is that manufacturer does not have full control over the isotope injection, and therefore they are wary of storing details in DICOM that they can not verify.

**Comment by CPernet** [comment]
@neurolabusc this is not completely true, I gave you access to a library of many PET phantoms (GE, Phillips and Siemens) -- we just haven't come around to making it public yet. It is partially true though because this is not real validation for DICOM (for ecat we make one from scratch)

**Comment by neurolabusc** [comment]
@CPernet my request is that the BIDS Specification Extensions be done in parallel and in collaboration with implementation considerations. I have consistently requested sample datasets from teams that have access hardware and licenses that I do not have direct access to. Where possible, I have worked to make publicly accessible validation datasets that allow all conversion tools to have concrete exemplars.

**Comment by CPernet** [comment]
+1 EEG native EDF from commercial software doesn't exist, does it? 

I can however relate to this as in PET some information is hidden in private tags that we do not necessarily know how to convert. It is thus not as much a spec issue as a conversion tool issue, on the other hand, if the spec say use nii and we don't have tools like Chris's dcm2niix then the spec is useless. 

Maybe a more general point is some BEP are asking for things but do not check where/if available for different data types/format - and that may reflect a difference in who leads which BEP.  Back to MEEG, leaders were also software like you who have read and converted native data for years so it was well thought out. As a 1st thing to do, we could update recommendations for BEP mentioning this issue.
